v1/feed.js

```diff
@@ -42,127 +87,111 @@ class Feed {
-            // of the components and store them
-            this._makeFeedRequests(Object.keys(FEED_URIS), hyper, rp, dateArr)
-            .then((result) => {
-                // assemble the final response to be returned
-                const finalResult = {
-                    status: 200,
-                    headers: {
-                        'cache-control': this.options.feed_cache_control,
-                        // mimic MCS' ETag value
-                        etag: `${dateArr.join('')}/${uuid.now().toString()}`,
-                        // TODO: need a way to dynamically derive this
-                        'content-type': 'application/json; charset=utf-8; ' +
-                        'profile="https://www.mediawiki.org/wiki/Specs/aggregated-feed/0.5.0"'
-                    },
-                    body: {}
-                };
-                // populate its body
-                Object.keys(result).forEach((key) => {
-                    if (result[key].body && Object.keys(result[key].body).length) {
-                        finalResult.body[key] = result[key].body;
-                    }
-                });
-                // store it
-                return hyper.put({
-                    uri: new URI([rp.domain, 'sys', 'key_value', 'feed.aggregated', date]),
-                    headers: finalResult.headers,
-                    body: finalResult.body
-                }).then(() => finalResult);
-            }))
-        .then((res) => {
-            // We've got the titles, populate them with summaries
-            const feed = res.body;
-            const summaries = {};
-            const requestTitle = (title) => {
-                if (title && !summaries[title]) {
-                    summaries[title] = hyper.get({
-                        uri: new URI([rp.domain, 'v1', 'page', 'summary', title])
-                    })
-                    .get('body')
-                    // Swallow the error, no need to fail the whole feed
-                    // request because of one failed summary fetch
-                    .catchReturn(undefined);
-                }
-            };
-
-            if (feed.tfa) {
-                requestTitle(feed.tfa.title);
-            }
-            if (feed.mostread && feed.mostread.articles) {
-                feed.mostread.articles.forEach((article) => { requestTitle(article.title); });
-            }
-            if (feed.news) {
-                feed.news.forEach((newsItem) => {
-                    if (newsItem.links) {
-                        newsItem.links.forEach((article) => {
-                            requestTitle(article.title);
-                        });
-                    }
-                });
+        const date = getDateSafe(rp);
+        const dateKey = toKey(date);
+        const dateArr = dateKey.split('-');
+        const populateSummaries = (res) => {
+            function fetchSummary(uri) {
+                return hyper.get({ uri })
```

Comment:
Wouldn't it be good to regex-test for the RB part in the URI, and if it matches, issue direct internal requests instead of going to Varnish and friends?